src/main/java/org/broadinstitute/hellbender/tools/dragstr/ComposeSTRTableFile.java

```diff
@@ -0,0 +1,269 @@
+                }
+            }
+            output.store(outputPath);
+        }
+        progressMeter.stop();
+    }
+
+    private Map<String, List<SimpleInterval>> composeAndGroupTraversalIntervalsByContig(final SAMSequenceDictionary dictionary) {
+        if (!intervalArgumentCollection.intervalsSpecified()) {
+            return dictionary.getSequences().stream()
+                    .map(s -> new SimpleInterval(s.getSequenceName(), 1, s.getSequenceLength()))
+                    .collect(Collectors.groupingBy(SimpleInterval::getContig, LinkedHashMap::new, Collectors.toList()));
+        } else {
+            final Map<String, List<SimpleInterval>> keyUnsorted = IntervalUtils.sortAndMergeOverlappingIntervals(intervalArgumentCollection.getIntervals(dictionary));
+            final Map<String, List<SimpleInterval>> keySorted = new LinkedHashMap<>(keyUnsorted.size());
+            keyUnsorted.entrySet().stream()
+                    .sorted(Comparator.comparingInt(entry -> dictionary.getSequenceIndex(entry.getKey())))
+                    .forEach(entry -> keySorted.put(entry.getKey(), entry.getValue()));
+            return keySorted;
+        }
+    }
+
+    private void initializeMasks(final SAMSequenceDictionary dictionary) {
+        nextMasks = new int[dictionary.getSequences().size()][maxPeriod + 1][maxRepeat + 1];
+        for (int i = 0; i < nextMasks.length; i++) {
+            for (final int[] masks : nextMasks[i]) {
+                Arrays.fill(masks, i);
+            }
+        }
+    }
+
+    private void traverseInterval(final int seqNumber, final BufferedReferenceBases sequence, final long seqStart, final long seqEnd,
+                                  final STRDecimationTable decimationTable, final STRTableFileBuilder output)
+    {
+        final String id = sequence.id();
+        final long length = sequence.length();
+        final byte[] unitBuffer = new byte[this.maxPeriod];
+        long pos = seqStart;
+        while (pos <= seqEnd) {
+           // usually maxPeriodAtPos == maxPeriod except when close to the end of the sequence
+           // is the maximum period to be considered that cannot exceed min(maxPeriod, seq-length - pos + 1)
+           // but is always 1 or greater.
+           final int maxPeriodAtPos = sequence.copyBytesAt(pos, unitBuffer, 0, maxPeriod);
+
+           // Efficient code for period == 1:
+           final byte firstUnitBase;
+           if (!Nucleotide.decode(firstUnitBase = unitBuffer[0]).isStandard()) {
+               pos++; continue;
+           }
+           long beg, end;
+
+           // we look upstream for same bases:
+           //noinspection StatementWithEmptyBody
+           for (beg = pos - 1; beg >= 1 && Nucleotide.same(sequence.byteAt(beg), firstUnitBase); beg--) { /* nothing to be done */ }
+           beg++;
+
+           // we look downstream for same bases:
+           //noinspection StatementWithEmptyBody
```

Comment:
Ok to the stub, the second not sure if is practical, will do if so.

Reply:
Actually I have to confess I'm not sure what you mean with the second part of your comment here. Since is not compulsory I will not do anything further unless you insist.